Sherpmail

AaryanHazCompter opened the pull request:
I saw an [issue Ian raised](https://github.com/Sooraj-beep/sherp2.0/issues/65) about wanting a way to (a) communicate to the server through the bot (b) have the server communicate to him through the bot; the idea seemed funny enough so I coded it - there's little error handling, no accounting for excess characters, no graphics and attachment handling, and definitely no spam prevention - however I am sleep deprived and wanted to make what is basically a letter mail system work to a reasonable extent.

I have never split PRs in my life I don't know what I did but I hope this is right lol

AaryanHazCompter committed:
add sherpmail

I saw an issue Ian raised about wanting a way to (a) communicate to the server through the bot (b) have the server communicate to him through the bot and the idea seemed funny so I coded it - there's little error handling, no accounting for excess characters, no graphics and attachment handling, and definitely no spam prevention - however I am sleep deprived and wanted to make what is basically a letter mail system work to a reasonable extent.

jadc committed:
Reaction events now fire on uncached messages (#86)

* fix: votedelete and starboard works on uncached messages

AaryanHazCompter committed:
Merge branch 'Sooraj-beep:master' into sherpmail

IanDeHaan commented:
OMG MERGE MERGE MERGE

Mattwmaster58 commented on cogs/sherpmail.py:
```diff
@@ -0,0 +1,68 @@
+
+    @commands.Cog.listener()
+    async def on_message(self, message: discord.Message):
+        if (message.guild == None) and (message.author.id == 214657501193306112):
```
```suggestion
        if (message.channel.type === 'dm') and (message.author.id == 214657501193306112):
```
I think this is clearer

AaryanHazCompter replied:
Adjusted to `message.channel.type.name == "private"` and it will reflect on the next commit.

Mattwmaster58 commented on cogs/sherpmail.py:
```diff
@@ -0,0 +1,68 @@
+        self.sherp = await interaction.client.fetch_user(212613981465083906)
+        try:
+            embed = discord.Embed(
+                title = f"📬 You've got mail! **@{interaction.user} | ID: {interaction.user.id}** sent you the following:",
```
This seems incorrect, unless discord.py has some special `__repr__` stuff I'm ignorning. I think we should mention interaction.user.global_name and interaction.user.name. The numeric ID isn't too useful IDT.

https://discordpy.readthedocs.io/en/stable/interactions/api.html?highlight=interaction#discord.Interaction.user

AaryanHazCompter replied:
The unique UID and user would prevent someone from writing stupid things as they would know that their identifiers were also being sent to Sherp. I did rewrite it and have it stated in the footer of the embed instead.

Mattwmaster58 commented on cogs/sherpmail.py:
```diff
@@ -0,0 +1,68 @@
+# Nonetheless it is nearly 6am and I felt the need to complete this
+# I hope it works out and nobody abuses it lol good luck
+
+class sherpMailbox(commands.Cog):
```
```suggestion
class SherpMailbox(commands.Cog):
```
(style)

AaryanHazCompter replied:
Good call! Applied to the incoming commit.

Mattwmaster58 reviewed (changes requested):
Looks good for the most part, have you tested this on yourself?

AaryanHazCompter commented:
> Looks good for the most part, have you tested this on yourself?

Yes, worked when I ran it on myself using my own channel, expect the effect to carry over to Sherp. I implemented a change you recommended regarding the appearance of UID in the title field, moved it to the footer.
![image](https://github.com/Sooraj-beep/sherp2.0/assets/44059614/83727c47-1b0f-4070-8342-c38ef89dc3cb)
![image](https://github.com/Sooraj-beep/sherp2.0/assets/44059614/196540d6-6783-470d-bd94-0170ada7d584)

AaryanHazCompter committed:
sherpmail modifications + init addition

Modified sherpmail cog from a cosmetic perspective based on inputs from Matt; added sherpmail cog to init file for setup and consistency

Sooraj-beep commented:
> OMG MERGE MERGE MERGE

Are you sure u want this? Could result in a lot of spam which you may not even be able to block (afaik)

AaryanHazCompter commented:
> > OMG MERGE MERGE MERGE
> 
> Are you sure u want this? Could result in a lot of spam which you may not even be able to block (afaik)

I do believe you can restrict specific application commands to specific channels - you can create a designated channel and set an extended slow mode timer on that chat which impacts individual users. Additionally, Ian can block and/or mute DMs  and it would be as if the bot was never texting him in the first place.

IanDeHaan commented:
Can this be merged

Sooraj-beep commented:
got some conflicts here

AaryanHazCompter committed:
Merge branch 'master' into sherpmail

AaryanHazCompter commented:
@Sooraj-beep @IanDeHaan Merge resolved should still work fine lmk I'll look at it again after finals

AaryanHazCompter committed:
The aura loss is insurmountable

I've got too much caffeine in my body to excuse this oversight
